website/docs/r/range.html.md

```diff
@@ -0,0 +1,64 @@
+---
+layout: "random"
+page_title: "Random: random_range"
+sidebar_current: "docs-random-resource-range"
+description: |-
+  Generates a random integer value from a range.
+---
+
+# random\_range
+
+The resource `random_range` generates random values from a given range, described by the `min` and `max` attributes of a given resource.
+
+This resource can be used in conjunction with resources that have
+the `create_before_destroy` lifecycle flag set, to avoid conflicts with
+unique names during the brief period where both the old and new resources
+exist concurrently.
+
+## Example Usage
+
+The following example shows how to generate a random priority between 1 and 99999 for
+a `aws_alb_listener_rule` resource:
+```hcl
+resource "random_range" "priority" {
+  min     = 1
+  max     = 99999
+  keepers = {
+    # Generate a new range each time we switch to a new listener ARN
+    listener_arn = "${var.listener_arn}"
+  }
+}
+
+resource "aws_alb_listener_rule" "main" {
+  listener_arn = "${var.listener_arn}"
+  priority     = "${random_range.priority.id}"
+
+  action {
+    type             = "forward"
+    target_group_arn = "${var.target_group_arn}"
+  }
+  # ... (other aws_alb_listener_rule arguments) ...
+}
+```
+
+The result of the above will set a random priority.
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `min` - (int) The minimum inclusive value of the range.
+
+* `max` - (int) The maximum inclusive value of the range.
+
+* `keepers` - (Optional) Arbitrary map of values that, when changed, will
+  trigger a new id to be generated. See
+  [the main provider documentation](../index.html) for more information.
+
+* `seed` - (Optional) A custom seed to always produce the same value.
+
+## Attribute Reference
+
+The following attributes are supported:
+
+* `id` - (int) The random range value.
```

Comment:
It feels a _little_ unintuitive to use `id` for this, since the result probably isn't an id in the usual sense we mean it.

We are required to call `SetId` to give Terraform something to track with, but I'd suggest that we consider that an undocumented implementation detail and instead publish a computed attribute called `result` which is documented here.

```hcl
  priority     = "${random_range.priority.result}"
```

What do you think? (I chose `result` here again for consistency with `random_string`)

Reply:
I tried to stick to the notation of already existing providers (e.g. pet provider) which also populate an `id` attribute as a result.

Regardless, I agree with you, `result` should be it! 👍